Frameworks/CoreText/CTFrame.mm

```diff
@@ -108,27 +108,46 @@ void CTFrameGetLineOrigins(CTFrameRef frameRef, CFRange range, CGPoint origins[]
     }
 }
 
-/**
- @Status Interoperable
-*/
-void CTFrameDraw(CTFrameRef frame, CGContextRef ctx) {
+void __CTFrameDraw(CTFrameRef frame, CGContextRef ctx, bool invertedCoordinates) {
     uint32_t count = [((_CTFrame*)frame)->_lines count];
-    CGPoint curTextPos = CGContextGetTextPosition(ctx);
     if (frame && ctx) {
         CGPoint curTextPos = CGContextGetTextPosition(ctx);
         for (_CTLine* line in static_cast<id<NSFastEnumeration>>(static_cast<_CTFrame*>(frame)->_lines)) {
             CGPoint newPos = curTextPos + line->_lineOrigin;
             CGContextSetTextPosition(ctx, newPos.x, newPos.y);
-            _CTLineDraw(static_cast<CTLineRef>(line), ctx, false);
+            _CTLineDraw(static_cast<CTLineRef>(line), ctx, false, invertedCoordinates);
         }
     }
 }
 
 /**
+ @Status Interoperable
+*/
+void CTFrameDraw(CTFrameRef frame, CGContextRef ctx) {
+    __CTFrameDraw(frame, ctx, true);
+}
+
+// Exposed to UIKit to draw without having to flip coordinate system or know size of bounding frame
+void CTFrameDrawUninverted(CTFrameRef frame, CGContextRef ctx) {
+    __CTFrameDraw(frame, ctx, false);
+}
+
+/**
  @Status Stub
  @Notes
 */
 CFTypeID CTFrameGetTypeID() {
     UNIMPLEMENTED();
     return StubReturn();
-}
\ No newline at end of file
+}
+
+// Private helper method for NSLayoutManager
+CTFrameRef CTFrameCreateWithAttributedString(CFAttributedStringRef attributedString) {
```

Comment:
>CTFrameCreateWithAttributedString [](start = 11, length = 33)

we should be following some sort of naming convention to identify this is not part of the public api. 
Looking at the name, makes me think it might be a public API.